src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerCommitImageFunctionalTest.groovy

```diff
@@ -0,0 +1,113 @@
+package com.bmuschko.gradle.docker.tasks.image
+
+import com.bmuschko.gradle.docker.AbstractGroovyDslFunctionalTest
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.UnexpectedBuildFailure
+
+class DockerCommitImageFunctionalTest extends AbstractGroovyDslFunctionalTest {
+
+    def "can commit image"() {
+        given:
+        String commitImage = """
+            task commitImage(type: DockerCommitImage) {
+                dependsOn startContainer
+                targetContainerId createContainer.getContainerId()
+                author = "john doe"
+                message = "My image created"
+                tag = "myimage:latest"
+            }
+        """
+
+        buildFile << containerStart(commitImage)
+
+        when:
+        BuildResult result = build('commitImage')
+
+        then:
+        result.output.contains("Commiting image for container")
+    }
+
+    def "cannot commit image without container"() {
+        given:
+        String commitImage = """
+            task commitImage(type: DockerCommitImage) {
+                dependsOn startContainer
+                finalizedBy removeContainer
+            }
+        """
+
+        buildFile << containerStart(commitImage)
+
+        when:
+        BuildResult result = build('commitImage')
```

Comment:
As a side note: You'd use `buildAndFail` in the case that you expect the build to run and fail. In those cases you won't need to catch an exception.

Reply:
Great! I did not know about that method.